src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/ConstructionTapeHelper.java

```diff
@@ -0,0 +1,289 @@
+        final BlockPos pos = workOrder.getBuildingLocation();
+        int tempRotation = 0;
+
+        if (workOrder.getRotation() == 0 && !(workOrder instanceof WorkOrderBuildDecoration))
+        {
+            final IBlockState blockState = world.getBlockState(pos);
+            if (blockState.getBlock() instanceof AbstractBlockHut)
+            {
+                tempRotation = BlockUtils.getRotationFromFacing(blockState.getValue(AbstractBlockHut.FACING));
+            }
+        }
+        else
+        {
+            tempRotation = workOrder.getRotation();
+        }
+
+        wrapper.rotate(tempRotation, world, workOrder.getBuildingLocation());
+        wrapper.setPosition(pos);
+        int x1 = wrapper.getPosition().getX() - wrapper.getOffset().getX() - 1;
+        int z1 = wrapper.getPosition().getZ() - wrapper.getOffset().getZ() - 1;
+        int x3 = wrapper.getPosition().getX() + (wrapper.getWidth() - wrapper.getOffset().getX());
+        int z3 = wrapper.getPosition().getZ() + (wrapper.getLength() - wrapper.getOffset().getZ());
+        int y = wrapper.getPosition().getY();
+        int newY;
+
+            if (x1 < x3)
+            {
+                for (int i = x1+1; i < x3; i++)
+                {
+                    newY = checkIfPlaceable(i,y,z1,world);
+                    BlockPos row1 = new BlockPos(i, newY, z1);
+                    world.setBlockState(row1, ModBlocks.blockConstructionTape.getDefaultState().withProperty(FACING, EnumFacing.SOUTH));
+                    newY = checkIfPlaceable(i,y,z3,world);
+                    BlockPos row2 = new BlockPos(i, newY, z3);
+                    world.setBlockState(row2, ModBlocks.blockConstructionTape.getDefaultState().withProperty(FACING, EnumFacing.NORTH));
+                }
+            }
+            else
+            {
+                for (int i = x3+1; i < x1; i++)
+                {
+                    newY = checkIfPlaceable(i,y,z1,world);
+                    BlockPos row1 = new BlockPos(i, newY, z1);
+                    world.setBlockState(row1, ModBlocks.blockConstructionTape.getDefaultState().withProperty(FACING, EnumFacing.SOUTH));
+                    newY = checkIfPlaceable(i,y,z3,world);
+                    BlockPos row2 = new BlockPos(i, newY, z3);
+                    world.setBlockState(row2, ModBlocks.blockConstructionTape.getDefaultState().withProperty(FACING, EnumFacing.NORTH));
+                }
+            }
+            if (z1 < z3)
+            {
+                for (int i = z1+1; i < z3; i++)
+                {
+                    newY = checkIfPlaceable(x1,y,i,world);
+                    BlockPos row3 = new BlockPos(x1, newY, i);
+                    world.setBlockState(row3, ModBlocks.blockConstructionTape.getDefaultState().withProperty(FACING, EnumFacing.EAST));
+                    newY = checkIfPlaceable(x3,y,i,world);
+                    BlockPos row4 = new BlockPos(x3, newY, i);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'row4' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)